src/zenml/utils/tag_utils.py

```diff
@@ -0,0 +1,636 @@
+#  Copyright (c) ZenML GmbH 2025. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+"""Utility functions for tags."""
+
+from typing import (
+    TYPE_CHECKING,
+    List,
+    Optional,
+    TypeVar,
+    Union,
+    overload,
+)
+from uuid import UUID
+
+from pydantic import BaseModel
+
+from zenml.enums import ColorVariants, TaggableResourceTypes
+from zenml.logger import get_logger
+
+add_tags_warning = """
+# Automatic tagging to a pipeline run (within a step)
+add_tags(tags=[...])
+
+# Manual tagging to a pipeline run
+add_tags(tags=[...], run=...)
+
+# Manual tagging to a pipeline
+add_tags(tags=[...], pipeline=...)
+
+# Manual tagging to a run template
+add_tags(tags=[...], run_template=...)
+
+# Manual tagging to an artifact
+add_tags(tags=[...], artifact=...)
+
+# Automatic tagging to an artifact version(within a step)
+add_tags(tags=[...], infer_artifact=True)  # step with single output
+add_tags(tags=[...], artifact_name=..., infer_artifact=True)  # specific output of a step
+
+# Manual tagging to an artifact version
+add_tags(tags=[...], artifact_name=..., artifact_version=...)
+add_tags(tags=[...], artifact_version_id=...)
+"""
+
+remove_tags_warning = """
+# Automatic tag removal from a pipeline run (within a step)
+remove_tags(tags=[...])
+
+# Manual tag removal from a pipeline run
+remove_tags(tags=[...], run=...)
+
+# Manual tag removal from a pipeline
+remove_tags(tags=[...], pipeline=...)
+
+# Manual tag removal from a run template
+remove_tags(tags=[...], run_template=...)
+
+# Manual tag removal from an artifact
+remove_tags(tags=[...], artifact=...)
+
+# Automatic tag removal from an artifact version (within a step)
+remove_tags(tags=[...], infer_artifact=True)  # step with single output
+remove_tags(tags=[...], artifact_name=..., infer_artifact=True)  # specific output of a step
+
+# Manual tag removal from an artifact version
+remove_tags(tags=[...], artifact_name=..., artifact_version=...)
+remove_tags(tags=[...], artifact_version_id=...)
+"""
+
+
+if TYPE_CHECKING:
+    from zenml.models import TagRequest
+    from zenml.zen_stores.schemas.base_schemas import BaseSchema
+
+    AnySchema = TypeVar("AnySchema", bound=BaseSchema)
+
+logger = get_logger(__name__)
+
+
+class Tag(BaseModel):
+    """A tag is a label that can be applied to a pipeline run."""
```

Comment:
`A tag is a label` why don't we call it label then? 🤔

I think a tag is just a tag

Reply:
Fixed it.